File: 03.CPP/0026.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int removeDuplicates(std::vector<int>& nums) {
        int low = 0;
        int high = 1;
        int uniqueCount = 1;
        int n = nums.size();

        while (high < n) {
            if (nums[high - 1] == nums[high]) {
                high++;
                continue;
            }

            nums[low + 1] = nums[high];
            high++;
            low++;
            uniqueCount++;
        }

        return uniqueCount;
    }
};

int main() {
    Solution sol;
    
    std::vector<int> nums = {1, 1, 1, 2, 2, 3};
    int result = sol.removeDuplicates(nums);
    std::cout << "Test 1: " << result << ", [";
    for (int i = 0; i < result; i++) {
        std::cout << nums[i] << (i < result - 1 ? ", " : "");
    }
    std::cout << "]" << std::endl;

    return 0;
}

File: 01.Python/0026.py
from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        low = 0
        high = 1
        uniqueCount = 1
        n = len(nums)

        while high < n:
            if nums[high] == nums[high - 1]:
                high += 1
                continue
            
            nums[low + 1] = nums[high]
            low += 1
            high += 1
            uniqueCount += 1
        
        return uniqueCount

sol = Solution()

nums = [1, 1, 1, 2, 2, 3]
result = sol.removeDuplicates(nums)

print(nums)
print(result)